### PR TITLE
Support QLR file upload format

### PR DESCRIPTION
**Pull request by lucernae**
Improvements to generate QGIS project from provided QLR file.
Useful for remote data source.

### CODE REVIEW

**Comment by lucernae** [comment]
This will enable easier integration for kartoza/geonode#422 . Will enable GeoSAFE to consume InaSAFE layers from remote data source (WFS, PostGIS, etc)

**Comment by NyakudyaA** [comment]
> This will enable easier integration for [kartoza/geonode#422](https://github.com/kartoza/geonode/issues/422) . Will enable GeoSAFE to consume InaSAFE layers from remote data source (WFS, PostGIS, etc)

Awesome @lucernae looking forward to test it out. Will it still be one layer per project ?

**Comment by lucernae** [comment]
@NyakudyaA yes one layer per project, due to Geonode layer models only recognized one layer file per upload.
If QLR works, we could do lot's of stuff with it, like provides multiple basemap/styles options in the same QLR file, or (the most interesting things) is to combine multiple layer file with different name and include it in one QLR. GeoNode will still thinks it is just one layer (one QLR file), but QGIS will be able to compose the layer in that QLR.
Or we can just extend that for GeoNode to import that QLR as GeoNode maps instead, but we need to provide the UI for QLR import (probably in Create Map view).

File: filters/tools.py [hunk 1]
```diff
@@ -145,6 +147,13 @@ def layer_from_source(source_uri, name):
         elif source_uri.endswith(raster_extensions):
             qgis_layer = QgsRasterLayer(sanitized_uri, name)
 
+        elif source_uri.endswith(qlr_extensions):
+
+            qgis_layer = QgsMapLayer.fromLayerDefinitionFile(sanitized_uri)
+            if qgis_layer:
```
Comment (Gustry) [inline filters/tools.py:153]: Missing the `isValid()`?

Comment (lucernae) [reply filters/tools.py:153]: Thanks @Gustry ! `isValid()` is tested in the outside in `map_composition.py` together with the error message to return. In here, I just check if the loader returns empty list or with loaded layer. I'm still not sure how to handle multiple layer, though. Maybe this one will do for now?

**Pull request merged**